RELEASE.md

```diff
@@ -13,18 +13,17 @@
         ```
       * Even if a submodule does not have a new release, it should be checked to see if there were any changes warranting cutting a new release and then including it.
     * Rotate the below release managers table placing yourself at the bottom. The date should be the first Wednesday of the month.
-2. Add all merged pull requests to the milestone for the release and create a new milestone for a next release e.g. `Release 1.16`.
```

Comment:
This section should be removed as well https://github.com/jaegertracing/jaeger/blob/main/CONTRIBUTING.md#merging-prs

Reply:
Thanks @pavolloffay; just to check, do you mean we should delete the entire "Merging PRs" section or just the last two points:

```
* pull request is assigned to the current release milestone
* add `changelog:*` and other labels
```

Reply:
just deleting these two bullets is fine